src/ng/rootScope.js

```diff
@@ -353,11 +353,13 @@ function $RootScopeProvider() {
        *    - `newVal` contains the current value of the `watchExpression`
        *    - `oldVal` contains the previous value of the `watchExpression`
        *    - `scope` refers to the current scope
-       * @param {boolean=} objectEquality Compare for object equality using {@link angular.equals} instead of
-       *     comparing for reference equality.
+       * @param {boolean=|function()} objectEquality Compare for object equality using {@link angular.equals} instead of
+       *     comparing for reference equality. If a function is passed, it will be used as a replacement for {@link angular.equals}.
+       * @param {function()} copier Replacement function for {@link angular.copy} to save value into oldVal for next
+            change event.
        * @returns {function()} Returns a deregistration function for this listener.
        */
-      $watch: function(watchExp, listener, objectEquality) {
+      $watch: function(watchExp, listener, objectEquality, copier) {
```

Comment:
Why is that ? What is wrong with using default `copy()` with a custom `objectEquality()` ?
I will usually not make sense, but why should it be our concern ? :grin: